### PR TITLE
Add missing val/test hooks in LightningModule

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/5442

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
Thanks @rohitgr7 , awesome!

File: tests/models/test_hooks.py [hunk 1]
```diff
@@ -329,16 +335,19 @@ def on_test_model_train(self):
...
+        'on_validation_end',
         'on_validation_model_train',
         'on_epoch_end',
         'on_train_epoch_end',
         'on_train_end',
         'on_fit_end',
+        'teardown',
     ]
 
     assert model.called == expected
```
Comment (rohitgr7) [inline tests/models/test_hooks.py:353]: shall this be updated the way it's done with callback using mock?

Comment (awaelchli) [reply tests/models/test_hooks.py:354]: that would be great (in a follow up pr). let me know if you need help

File: pytorch_lightning/core/hooks.py [hunk 2]
```diff
@@ -84,6 +85,18 @@ def on_train_end(self) -> None:
         """
         # do something at the end of training
 
+    def on_validation_start(self):
```
Comment (SkafteNicki) [inline pytorch_lightning/core/hooks.py:88]: just to be consistent :]
```suggestion
    def on_validation_start(self) -> None:
```

File: pytorch_lightning/core/hooks.py [hunk 3]
```diff
@@ -84,6 +85,18 @@ def on_train_end(self) -> None:
...
+        """
+        Called at the beginning of validation.
+        """
+        # do something at the start of validation
+
+    def on_validation_end(self):
```
Comment (SkafteNicki) [inline pytorch_lightning/core/hooks.py:94]:
```suggestion
    def on_validation_end(self) -> None:
```

File: pytorch_lightning/core/hooks.py [hunk 4]
```diff
@@ -253,6 +266,18 @@ def on_test_epoch_end(self) -> None:
         """
         # do something when the epoch ends
 
+    def on_test_start(self):
```
Comment (SkafteNicki) [inline pytorch_lightning/core/hooks.py:269]:
```suggestion
    def on_test_start(self) -> None:
```

File: pytorch_lightning/core/hooks.py [hunk 5]
```diff
@@ -253,6 +266,18 @@ def on_test_epoch_end(self) -> None:
...
+        """
+        Called at the beginning of testing.
+        """
+        # do something at the start of testing
+
+    def on_test_end(self):
```
Comment (SkafteNicki) [inline pytorch_lightning/core/hooks.py:275]:
```suggestion
    def on_test_end(self) -> None:
```

**Review by SkafteNicki — APPROVED** [review]
LGTM

File: pytorch_lightning/core/hooks.py (hunk 2 above)
Comment (rohitgr7) [reply pytorch_lightning/core/hooks.py:88]: ah thanks :+1:

File: tests/models/test_hooks.py (hunk 1 above)
Comment (rohitgr7) [reply tests/models/test_hooks.py:354]: sure :+1:

**Review by carmocca — APPROVED** [review]

**Review by williamFalcon — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Pull request merged**